### PR TITLE
JAMES-4044 LDAP matchers: support extracting attribute from DN

### PR DESCRIPTION
**Pull request by quantranhong1999**
Supports extracting LDAP attribute from its DN value. e.g.

`description: cn=active,ou=accountStatus,ou=whatever,o=gov,c=vn` => `description: active`

- [x] HasLDAPAttribute
- [x] SenderHasLDAPAttribute
- ...

### CODE REVIEW

File: server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/HasLDAPAttribute.java [hunk 1]
```diff
@@ -170,10 +172,27 @@ private boolean hasAttribute(SearchResultEntry entry) {
...
+    public static String extractLdapAttributeValue(String ldapValue) {
+        if (ldapValue.contains(",")) {
+            try {
+                return Arrays.stream(new DN(ldapValue).getRDNs())
+                    .flatMap(rdn -> rdn.getNameValuePairs().stream())
+                    .filter(pair -> pair.getAttributeName().equals("cn"))
+                    .findFirst()
+                    .map(RDNNameValuePair::getAttributeValue)
+                    .orElse(ldapValue);
+            } catch (LDAPException e) {
```
Comment (chibenwa) [inline server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/HasLDAPAttribute.java:189]: INFO log with the actual value in there.

File: server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/HasLDAPAttribute.java [hunk 2]
```diff
@@ -170,10 +176,28 @@ private boolean hasAttribute(SearchResultEntry entry) {
         return attributeValue.map(value -> Optional.ofNullable(entry.getAttribute(attributeName))
                 .map(attribute -> Arrays.stream(attribute.getValues()))
                 .orElse(Stream.empty())
+                .map(ldapValue -> extractLdapAttributeValue(ldapValue, attributeName, LOGGER))
```
Comment (chibenwa) [inline server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/HasLDAPAttribute.java:179]: Why passing the LOGGER as a parameter?

File: server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/HasLDAPAttribute.java [hunk 3]
```diff
@@ -170,10 +176,28 @@ private boolean hasAttribute(SearchResultEntry entry) {
...
                 .anyMatch(value::equals))
             .orElseGet(() -> entry.hasAttribute(attributeName));
     }
 
+    public static String extractLdapAttributeValue(String ldapValue, String attributeName, Logger logger) {
```
Comment (chibenwa) [inline server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/HasLDAPAttribute.java:184]: Could be better suited in a different class named AttributeUtils

Comment (chibenwa) [inline server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/HasLDAPAttribute.java:184]: I'd put attributeName first and would not be against of strong typing

**Review by Arsnael — APPROVED** [review]

**Pull request merged**